src/sdk/index.ts

```diff
@@ -419,36 +343,97 @@ export async function createPlayground(
+    } else {
+      playgroundUrl.searchParams.set('lite', 'true');
+    }
+  }
+  if (view) {
+    // eslint-disable-next-line no-console
+    console.warn(
+      `Deprecation notice: The "view" option has been moved to "config.view". For headless mode use "headless: true".`,
+    );
+    if (typeof config === 'object' && config.view == null && view !== 'headless') {
+      config.view = view;
+    } else {
+      playgroundUrl.searchParams.set('view', view);
+    }
+  } else if (typeof config === 'object') {
+    if (Object.keys(config).length > 0) {
+      playgroundUrl.searchParams.set('config', 'sdk');
+    }
+  } else {
+    throw new Error(`"config" is not a valid URL or configuration object.`);
+  }
+
+  if (typeof config === 'string') {
+    try {
+      new URL(config);
+      playgroundUrl.searchParams.set('config', encodeURIComponent(config));
+    } catch {
+      throw new Error(`"config" is not a valid URL or configuration object.`);
+    }
+  } else if (config && typeof config === 'object' && Object.keys(config).length > 0) {
+    playgroundUrl.searchParams.set('config', 'sdk'); // fixme: not sure of this one
+    if (config.title && config.title !== 'Untitled Project') {
+      playgroundUrl.searchParams.set('title', config.title);
+    }
+    if (config.description && config.description.length > 0) {
+      playgroundUrl.searchParams.set('description', config.description);
+    }
+    const hashParams = new URLSearchParams();
+    hashParams.set('x', 'code/' + compressToEncodedURIComponent(JSON.stringify(config)));
+    playgroundUrl.hash = hashParams.toString();
+  }
+
+  // handle params
   if (params && typeof params === 'object') {
+    // TODO: maybe store these in hash if they have an x param?
     try {
-      encodedParams = compressToEncodedURIComponent(JSON.stringify(params));
+      searchParams.set('params', compressToEncodedURIComponent(JSON.stringify(params)));
     } catch {
       (Object.keys(params) as Array<keyof UrlQueryParams>).forEach((param) => {
-        (params as any)[param] = encodeURIComponent(String(params[param]));
+        searchParams.set(param, encodeURIComponent(String(params[param])));
       });
     }
   }
 
-  const allParams = new URLSearchParams(
-    JSON.parse(
-      JSON.stringify({
-        ...otherOptions,
-        ...(encodedParams ? { params: encodedParams } : params),
-        ...{ x },
-        ...configParam,
-      }),
-    ),
-  ).toString();
-  return (appUrl || 'https://livecodes.io') + (allParams ? '?' + allParams : '');
+  if (template) {
+    playgroundUrl.searchParams.set('template', template);
+  }
+  if (importFrom) {
+    playgroundUrl.searchParams.set('x', encodeURIComponent(importFrom)); // TODO: how do we handle both a config and an importFrom
+  }
+  if (isHeadless) {
+    playgroundUrl.searchParams.set('headless', 'true');
+  }
+
+  playgroundUrl.searchParams.set('embed', 'true');
+  playgroundUrl.searchParams.set('loading', isHeadless ? 'eager' : loading);
```

Comment:
This should only be set if it was provided.
Note that we should not give it a default value here.

In `createPlayground` it should have the default value and set to param.